### PR TITLE
BWS-96: + added table for select permissions

### PR DESCRIPTION
**Pull request by bohdanshcherba**

### CODE REVIEW

File: frontend/src/components/eam-configurate-group/components/workers-table/helpers/get-columns.helper.tsx [hunk 1]
```diff
@@ -1,23 +1,18 @@
 import { UsersTableHeader, UsersTableAccessor } from 'common/enums/enums';
 import { Column } from 'react-table';
-import { SelectRowCell } from './cells/cells';
+import { SelectRowCell } from 'components/eam-configurate-group/helpers/helpers';
```
Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/components/workers-table/helpers/get-columns.helper.tsx]: use relative import

File: frontend/src/components/eam-configurate-group/eam-configurate-group.module.scss [hunk 2]
```diff
@@ -64,10 +64,6 @@
   max-width: 50%;
 }
 
-.table {
-  padding: 0;
-}
```
Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/eam-configurate-group.module.scss]: why?
attache the screenshot how it looks like

File: frontend/src/components/eam-configurate-group/eam-configurate-group.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import React, { FC, useMemo } from 'react';
-import { Button, Input, Table } from 'components/common/common';
+import React, { FC } from 'react';
```
Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/eam-configurate-group.tsx]:
```suggestion
import { FC } from 'react';
```

File: frontend/src/store/eam-group-configurate/reducer.ts [hunk 4]
```diff
@@ -20,6 +25,10 @@ const reducer = createReducer(initialState, (builder) => {
     state.dataStatus = DataStatus.FULFILLED;
     state.workers = action.payload.items;
   });
+  builder.addCase(getPermission.fulfilled, (state, action) => {
+    state.dataStatus = DataStatus.FULFILLED;
```
Comment (what1s1ove) [inline frontend/src/store/eam-group-configurate/reducer.ts]: we should have data status for each entity for loading (ex: PermissionsDataStatus)

File: frontend/src/components/eam-configurate-group/eam-configurate-group.tsx [hunk 5]
```diff
@@ -28,58 +28,56 @@ const EAMConfigurateGroup: FC = () => {
...
+  const handleRemoveWorkerId = (id: string): void => {
+    setSelectedWorkers((prevState) => prevState.filter((it) => it !== id));
+  };
+  const handleIsCheckedWorkerId = (id: string): boolean => {
+    return selectedWorkers.some((it) => it === id);
+  };
 
-  const data = useMemo(() => getRows(workers), [workers]);
+  const handleAddPermissionId = (id: string): void => {
+    setSelectedPermissions((prevState) => prevState.concat(id));
```
Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/eam-configurate-group.tsx]: duplicate in logic. create a custom hook for it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/eam-configurate-group/eam-configurate-group.module.scss (hunk 2 above)
Comment (bohdanshcherba) [reply frontend/src/components/eam-configurate-group/eam-configurate-group.module.scss]: I put this style in a folder for each table

File: frontend/src/components/eam-configurate-group/hooks/use-selected-items/use-selected-items.hook.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { useState } from 'hooks/hooks';
+import { useSelectedItemsType } from '../../common/types';
+
+const useSelectedItems = (): useSelectedItemsType => {
```
Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/hooks/use-selected-items/use-selected-items.hook.ts]: actually we will use this type in several places. move  it to the src/hooks

Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/hooks/use-selected-items/use-selected-items.hook.ts]:
```suggestion
type UseSelectedItemsHook<T> = () => {
  selectedItems: T[];
  handleAdd: (id: string) => void;
  handleRemove: (id: string) => void;
  handleCheck: (id: string) => boolean;
}

const useSelectedItems = <T>(items: T[] ): UseSelectedItemsResult<T> => {
```

File: frontend/src/components/eam-configurate-group/components/workers-table/helpers/get-columns.helper.tsx (hunk 1 above)
Comment (what1s1ove) [reply frontend/src/components/eam-configurate-group/components/workers-table/helpers/get-columns.helper.tsx]: isn't fixed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/eam-configurate-group/hooks/use-selected-items/use-selected-items.hook.ts (hunk 6 above)
Comment (what1s1ove) [reply frontend/src/components/eam-configurate-group/hooks/use-selected-items/use-selected-items.hook.ts]: isn't fixed

File: frontend/src/common/types/app/use-selected-items-hook.type.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+type UseSelectedItemsHook = {
+  selected: string[];
```
Comment (what1s1ove) [inline frontend/src/common/types/app/use-selected-items-hook.type.ts]: isn't fixed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

File: frontend/src/components/eam-configurate-group/eam-configurate-group.tsx [hunk 8]
```diff
@@ -18,69 +18,48 @@ import { getNameOf } from 'helpers/helpers';
 import { EAMGroupConfigurate as EAMGroupConfigurateActions } from 'store/actions';
 import styles from './eam-configurate-group.module.scss';
 import { EAMGroupConfigurateRequestDto } from 'common/types/types';
-import { getRows, getColumns } from './helpers/helpers';
-import { eamGroupConfigurate as CreateGroupValidationSchema } from 'validation-schemas/validation-schemas';
+import { eamGroupConfigurate } from 'validation-schemas/validation-schemas';
```
Comment (what1s1ove) [inline frontend/src/components/eam-configurate-group/eam-configurate-group.tsx]:
```suggestion
import { eamGroupConfigurate as eamGroupConfigurateValidationSchema } from 'validation-schemas/validation-schemas';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**